src/Compilers/CSharp/Test/Semantic/Semantics/TargetTypedConditionalOperatorTests.cs

```diff
@@ -0,0 +1,246 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Collections;
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests
+{
+    /// <summary>
+    /// Test binding of the target-typed conditional (aka ternary) operator.
+    /// </summary>
+    public class TargetTypedConditionalOperatorTests : CSharpTestBase
+    {
+        [Fact]
+        public void TestImplicitConversions_Good()
```

Comment:
nit: it seems like this test contains some "bad" cases, such as the first case that implicitly converts int to short. Was that intended?

Reply:
Those are tests that will become "good" tests if we permit target typing even when there is a natural type.  I think that change is likely after we discuss it at an upcoming LDM.  This is the compatible version of the spec until we have that discussion.

---
In reply to: [410355611](https://github.com/dotnet/roslyn/pull/40604#discussion_r410355611) [](ancestors = 410355611)